Makefile

```diff
@@ -407,16 +411,15 @@ tst : ;@$(MAKE) $(MFLAGS) V=$(V) test
 ctc-% : $(ctests);@$(foreach tst,$(ctests),$(tst) /cpu/$*;)
 
 prove : BACKENDS += $(TEST_BACKENDS)
-prove : $(matched_prereq)
+prove : $(matched)
 	$(info Testing backends: $(BACKENDS))
 	$(PROVE) $(PROVE_OPTS) --exec 'tests/tap.sh' $(matched:$(OBJDIR)/%=%)
 # run prove target in parallel
 prv : ;@$(MAKE) $(MFLAGS) V=$(V) prove
 
-prepnektests:
-	(export CC FC && cd examples && make prepnektests)
-
+.ONESHELL :
 prove-all :
+	+export NEK5K_DIR='$(NEK5K_DIR)' CEED_DIR='$(PWD)'
```

Comment:
@jedbrown @thilinarmtb how about this approach to pass `NEK5K_DIR` and `CEED_DIR` to the shell where we test

Reply:
Looks good.

Reply:
I think each of the lines run in different shells. Also, there are other targets to run nek tests. We can export those variables globally.

Reply:
I added `.OHESHELL` to get all of the lines in the same shell.

Reply:
And now bp1/bp3 is failing to build on Travis. Investigating.

Reply:
That can have other consequences. I'd rather [export](https://www.gnu.org/software/make/manual/html_node/Variables_002fRecursion.html) those variables specifically.

Reply:
I thought that was what the highlighted line does, but the following line ignored the export without `.ONESHELL`.

Reply:
The `+` says "the thing you're running behaves like make; it may use the jobserver, for example". The command is run in the shell. `export` is also a make keyword (see my link).